src/sentry/notifications/helpers.py

```diff
@@ -745,12 +741,13 @@ def get_all_setting_providers(
         user_ids: The user ids to get notification settings for.
     """
 
-    query = get_query(recipients, project, organization)
+    projects = [project] if project else []
+    query = get_query(recipients, projects, organization)
     return NotificationSettingProvider.objects.filter(query)
 
 
 def get_all_setting_options(
-    recipients: Iterable[RpcActor | Team | RpcUser] | None = None,
+    recipients: Iterable[RpcActor | Team | User] | None = None,
```

Comment:
Do the functions that have their signatures changed no longer run in region silo code? There won't be a way to get `User` in issue alerts or metric alert logic, only `RpcUser` is available there.